### PR TITLE
Add host-spawn

### LINKED ISSUE #315: "bash: cannot set terminal process group (-1): Inappropriate ioctl for device" when open Visual Studio Code integrated terminal
# Linux distribution and kernel version
Arch Linux x86_64 
5.17.7-arch1-1

# Flatpak version
Flatpak 1.12.7

# Description of the problem
When I configure the integrated terminal according to the flatpak_warning.txt in the config file located in `~/.var/app/com.visualstudio.code/config/Code/User/settings.json`
```
{
  "terminal.integrated.defaultProfile.linux": "bash",
  "terminal.integrated.profiles.linux": {
    "bash": {
      "path": "/usr/bin/flatpak-spawn",
      "args": ["--host", "--env=TERM=xterm-256color", "bash"]
    }
  }
}
```
I get the following message every time I open the integrated terminal
```
bash: cannot set terminal process group (-1): Inappropriate ioctl for device
bash: no job control in this shell
```
image

### PR DESCRIPTION
**Pull request by ran-dall**
Closes #315 #197 #62

This commits add `host-spawn` to this Flatpak in `/app/bin`.

Also, updates the `README.md` with the options to use `host-spawn`.

### CODE REVIEW

**Comment by lbssousa** [comment]
While this PR is not merged, one can install `host-spawn` binary to directory `$HOME/.var/app/com.visualstudio.code/data/node_modules/bin` (it’s already in VSCode Flatpak’s `PATH`)

**Comment by lbssousa** [comment]
Wouldn't it be better to package this as an add-on for VSCode Flatpak (e.g. `com.visualstudio.code.tool.host-spawn`)?

Could it be possible to ship separate add-ons for providing `host-spawn` symlinks to notorious binaries (e.g. `com.visualstudio.code.tool.host-spawn.podman`, `com.visualstudio.code.tool.host-spawn.fish`, `com.visualstudio.code.tool.host-spawn.bash`, etc.)?

**Comment by mdmundo** [comment]
@lbssousa i made a [pr](https://github.com/flathub/flathub/pull/4267)

File: README.md [hunk 1]
```diff
@@ -38,11 +43,27 @@ And make sure that you have the following lines there
...
 }
 ```
-* You can change **bash** to any terminal you are using: zsh, fish, sh.
+
+`host-spawn`
+
+```
+    "terminal.integrated.defaultProfile.linux": "bash",
+    "bash": {
+      "path": "/app/bin/host-spawn",
```
Comment (alatiera) [inline README.md:58]:
```suggestion
      "path": "/app/tools/host-spawn/bin/host-spawn",
```

**Comment by ran-dall** [comment]
Big thank you to @alatiera, @bilelmoussaoui, and everyone else that participated in getting this merged!

**Comment by lbssousa** [comment]
Please update host-spawn to version 1.4.2. It's needed to spawn host-system podman properly.

**Comment by alatiera** [comment]
> Please update host-spawn to version 1.4.2. It's needed to spawn host-system podman properly.

Wanna file a PR? Bumping the version and changing the sha should be everything needed hopefully :)

**Pull request merged**